ppx-lib/gen_js_api_ppx.ml

```diff
@@ -1219,7 +1227,15 @@ and gen_funs ~global_attrs p =
     match p.ptype_manifest, p.ptype_kind with
     | Some ty, Ptype_abstract ->
         let ty = parse_typ ~global_attrs ty in
-        lazy (js2ml_fun ty), lazy (ml2js_fun ty)
+        (match p.ptype_params, ty with
+        | [], _ ->
+          lazy (js2ml_fun ty), lazy (ml2js_fun ty)
+        | [{ptyp_desc = Ptyp_var _; ptyp_loc = _; ptyp_attributes = _; ptyp_loc_stack = _}, Invariant], Js ->
+          let s = "_" in  
+          lazy (fun_ (Nolabel, s) (js2ml_fun ty)),
+          lazy (fun_ (Nolabel, s) (ml2js_fun ty))
+        | _ ->
+          error p.ptype_loc Cannot_parse_type)
     | None, Ptype_abstract ->
```

Comment:
Side note. I realized this branch actually never runs, because `gen_funs` is called after processing the declarations through `rewrite_typ_decl`:

https://github.com/LexiFi/gen_js_api/blob/4e0f2a344a4d5598619982318ec841b77f923781/ppx-lib/gen_js_api_ppx.ml#L1335

And `rewrite_typ_decl` adds a manifest for this specific case:

https://github.com/LexiFi/gen_js_api/blob/4e0f2a344a4d5598619982318ec841b77f923781/ppx-lib/gen_js_api_ppx.ml#L743